fenix/app/src/test/java/org/mozilla/fenix/gleanplumb/state/MessagingMiddlewareTest.kt

```diff
@@ -178,27 +195,36 @@ class MessagingMiddlewareTest {
     }
 
     @Test
-    fun `WHEN onMessageDismissed THEN updateMetadata,removeMessage , UpdateMessages and removeMessageToShowIfNeeded`() = runTestOnMain {
+    fun `WHEN onMessageDismissed THEN updateMetadata,removeMessage , UpdateMessages and consumeMessageToShowIfNeeded`() = runTestOnMain {
         val message = Message(
             "control-id",
             mockk(relaxed = true),
             action = "action",
             mockk(relaxed = true),
             listOf("trigger"),
-            Message.Metadata("same-id"),
+            Message.Metadata("same-id", displayCount = 0),
+        )
+        val controller = NimbusMessagingController(messagingStorage) { 0 }
+        val store = AppStore(
+            AppState(
+                messaging = MessagingState(
+                    messages = listOf(
+                        message,
+                    ),
+                ),
+            ),
+            listOf(
+                MessagingMiddleware(messagingStorage, controller, coroutineScope),
+            ),
         )
 
-        val spiedMiddleware = spyk(middleware)
-
-        every { spiedMiddleware.removeMessage(middlewareContext, message) } returns emptyList()
-        every { spiedMiddleware.consumeMessageToShowIfNeeded(middlewareContext, message) } just Runs
-
-        spiedMiddleware.onMessageDismissed(middlewareContext, message)
+        store.dispatch(MessageDismissed(message))
+        store.waitUntilIdle()
 
-        coVerify { messagingController.onMessageDismissed(message.metadata) }
-        coVerify { messagingStorage.updateMetadata(message.metadata.copy(dismissed = true)) }
-        verify { middlewareContext.dispatch(UpdateMessages(emptyList())) }
-        verify { spiedMiddleware.removeMessage(middlewareContext, message) }
+        // removeMessages causes messages size to be 0
+        assertEquals(0, store.state.messaging.messages.size)
+        // consumeMessageToShowIfNeeded causes messageToShow map size to be 0
+        assertEquals(0, store.state.messaging.messageToShow.size)
```

Comment:
The assertion here is right, but if you look up at your state, you start off with `messageToShow` as empty. So we're asserting here that nothing has changed? Shouldn't we have had a `messageToShow` first, then after the dismissal, the message is removed from `messages` and `messageToShow`?

Reply:
Let's add an assertion before the `dispatch(MessageDismissed(message))` that the `messageToShow` has a message in there first.

Reply:
Yes, totally agree here